### PR TITLE
Generate application.properties when creating a project

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: devtools/common/src/main/java/io/quarkus/BasicRest.java [hunk 1]
```diff
@@ -167,8 +167,8 @@ private void createDockerFile() throws IOException {
     }
 
     private void createMicroProfileConfig() throws IOException {
-        File meta = new File(projectRoot, "src/main/resources/META-INF");
-        File file = new File(mkdirs(meta), "microprofile-config.properties");
+        File meta = new File(projectRoot, "src/main/resources");
+        File file = new File(mkdirs(meta), "application.properties");
```
Comment (gsmet) [inline devtools/common/src/main/java/io/quarkus/BasicRest.java]: Good catch. Maybe we should rename the method to `createApplicationConfig`?

Comment (geoand) [reply devtools/common/src/main/java/io/quarkus/BasicRest.java]: Even better catch :P, We certainly should!

Comment (geoand) [reply devtools/common/src/main/java/io/quarkus/BasicRest.java]: Fixed

File: devtools/common/src/main/java/io/quarkus/BasicRest.java [hunk 2]
```diff
@@ -167,8 +167,8 @@ private void createDockerFile() throws IOException {
...
         if (!file.exists()) {
             Files.write(file.toPath(), Arrays.asList("# Configuration file", "key = value"), StandardOpenOption.CREATE_NEW);
```
Comment (gsmet) [inline devtools/common/src/main/java/io/quarkus/BasicRest.java]: Unrelated but I think maybe we should add a `# ` before `key = value`. If at some point, we want to check the configuration is properly consumed, we will have this property defined in all the generated projects.

Comment (geoand) [reply devtools/common/src/main/java/io/quarkus/BasicRest.java]: That's a great idea! I'll add it

Comment (geoand) [reply devtools/common/src/main/java/io/quarkus/BasicRest.java]: Done

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
@geoand looks like someone forgot to run the tests :).

```
[ERROR] Failures: 
[ERROR]   CreateProjectTest.createNewWithCustomizations:140 
Expecting file:
  </home/vsts/work/1/s/devtools/common/target/existing/src/main/resources/META-INF/microprofile-config.properties>
to exist.
[ERROR]   CreateProjectTest.createOnTopPomWithResource:100 
Expecting file:
  </home/vsts/work/1/s/devtools/common/target/existing/src/main/resources/META-INF/microprofile-config.properties>
to exist.
[ERROR]   CreateProjectTest.createOnTopPomWithoutResource:58 
Expecting file:
  </home/vsts/work/1/s/devtools/common/target/existing/src/main/resources/META-INF/microprofile-config.properties>
```

**Comment by geoand** [comment]
@gsmet guilty as changed :P

Fixing now

**Comment by gsmet** [comment]
@geoand if possible, please squash the test changes with the other changes so that we have something atomic. Thanks!

**Comment by geoand** [comment]
@gsmet Fixed and squashed. Hopefully the tests will pass this time :O

**Comment by geoand** [comment]
There is saying here in Greece: when you want to avoid doing something you usually end up doing more work then you would have had you just done it in the first place.

In this case if I had run the tests the first time I would have been done with it a lot faster :)

**Comment by geoand** [comment]
@gsmet Tests are passing now

**Comment by gsmet** [comment]
Merged, thanks!

**Comment by geoand** [comment]
You're welcome :)

**Pull request merged**